File: app/src/main/java/com/example/pcmall/service/ImageService.java
package com.example.pcmall.service;

import com.example.pcmall.model.response.ResponseResult;

import io.reactivex.rxjava3.core.Observable;
import okhttp3.MultipartBody;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;

public interface ImageService {
    @Multipart
    @POST("upload")
    Observable<ResponseResult<String>> upload(@Part MultipartBody.Part file);
}

File: core/network/src/main/java/com/example/pcmallcompose/core/network/service/ImageService.kt
package com.example.pcmallcompose.core.network.service

import com.example.pcmallcompose.core.model.response.ResponseResult
import retrofit2.http.GET

interface ImageService {
    @GET("getADImageList")
    suspend fun getADImageList(): ResponseResult<List<String>>
}